src/repository/sessions.rs

```diff
@@ -0,0 +1,103 @@
+use axum::async_trait;
+use mockall::automock;
+use sqlx::PgPool;
+use tracing::error;
+
+use crate::domain::sessions::{SessionData, Session};
+
+pub enum SessionGetError {
+    Missing,
+    Unknown
+}
+
+pub enum SessionDeleteError {
+    Unknown
+}
+
+pub enum SessionInsertError {
+    Duplicate,
+    Unknown
+}
+
+#[automock]
+#[async_trait]
+pub trait SessionRepository {
+    async fn insert(&self, session: &SessionData) -> Result<(), SessionInsertError>;
+    async fn get(&self, id: &str) -> Result<Session, SessionGetError>;
+    async fn delete(&self, id: &str) -> Result<(), SessionDeleteError>;
+}
+
+#[derive(Debug, Clone)]
+pub struct PgSessionRepository {
+    pub pool: PgPool
+}
+
+impl PgSessionRepository {
+    pub fn new(pool: &PgPool) -> Self {
+        Self { pool: pool.clone() }
+    }
+}
+
+#[async_trait]
+impl SessionRepository for PgSessionRepository {
+    #[tracing::instrument(skip_all, fields(id = session_data.id))]
```

Comment:
We shouldn't log session id

Reply:
Indeed, I thought I removed it everywhere